de.metas.contracts/src/main/java/de/metas/contracts/impl/ContractChangeBL.java

```diff
@@ -86,31 +86,41 @@ public void cancelContract(@NonNull final I_C_Flatrate_Term currentTerm,
 			cancelContractIfNotCanceledAlready(initialContract, contractChangeParameters);
 		}
 	}
-	
+
+	private boolean isVoidSingleContract(final @NonNull ContractChangeParameters contractChangeParameters)
+	{
+		return ChangeTerm_ACTION_VoidSingleContract.equals(extractAction(contractChangeParameters));
+	}
+
+	private String extractAction(final @NonNull ContractChangeParameters contractChangeParameters)
```

Comment:
i think this is too much abstraction.... we don't need this method. More, it's used in only one place.